src/HttpClient.php

```diff
@@ -0,0 +1,59 @@
+<?php namespace DocRaptor;
+
+use DocRaptor\Exception\BadRequestException;
+use DocRaptor\Exception\ForbiddenException;
+use DocRaptor\Exception\UnauthorizedException;
+use DocRaptor\Exception\UnexpectedValueException;
+use DocRaptor\Exception\UnprocessableEntityException;
+use Exception;
+
+/**
+ * Class HttpClient
+ * @package DocRaptor
+ */
+class HttpClient implements HttpTransferInterface
+{
+    /**
+     * @param string $uri
+     * @param array $postFields
+     * @return mixed
+     * @throws Exception
+     */
+    public function doPost($uri, array $postFields)
```

Comment:
Ok, as I wrote above, these two are not possible, but the same lazy instantiation can be achieved with what I showed above, as you actually describe it in your last sentence.